components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ServiceCatalogDAO.java

```diff
@@ -448,4 +453,63 @@ public void deleteService(String serviceKey, int tenantId) throws APIManagementE
             handleException("Failed to delete service : " + serviceKey + " from service catalog: " + tenantId, e);
         }
     }
+
+    /**
+     * Get services
+     * @param filterParams Service Filter parameters
+     * @param tenantId Tenant ID of the logged in user
+     * @param shrink Whether to shrink the response or not
+     * @return List of Services
+     * @throws APIManagementException
+     */
+    public List<ServiceEntry> getServices(ServiceFilterParams filterParams, int tenantId, boolean shrink)
+            throws APIManagementException {
+        List<ServiceEntry> serviceEntryList = new ArrayList<>();
+        String query = SQLConstantManagerFactory.getSQlString("GET_ALL_SERVICES_BY_TENANT_ID");
+        query = query.replace("$1", filterParams.getSortBy());
+        query = query.replace("$2", filterParams.getSortOrder());
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement ps = connection.prepareStatement(query)) {
+            ps.setInt(1, tenantId);
+            ps.setString(2, "%" + filterParams.getName() + "%");
+            ps.setString(3, "%" + filterParams.getVersion() + "%");
+            ps.setString(4, "%" +filterParams.getDefinitionType() + "%");
+            ps.setString(5, "%" + filterParams.getDisplayName() + "%");
```

Comment:
formatting issue

Reply:
Fixed in 6b2fb36b46edd6b00e450e671b588625420420d2